cosmic-applet-graphics/src/window.rs

```diff
@@ -178,93 +143,199 @@ impl Application for Window {
         Command::none()
     }
 
     fn view(&self, id: SurfaceIdWrapper) -> Element<Message> {
         match id {
             SurfaceIdWrapper::LayerSurface(_) => unimplemented!(),
-            SurfaceIdWrapper::Window(_) => self
-                .applet_helper
-                .icon_button("input-gaming-symbolic")
-                .on_press(Message::TogglePopup)
-                .style(Button::Text)
-                .into(),
+            SurfaceIdWrapper::Window(_) => match self.applet_helper.anchor {
+                PanelAnchor::Left | PanelAnchor::Right => self
+                    .applet_helper
+                    .icon_button("input-gaming-symbolic")
+                    .on_press(Message::TogglePopup)
+                    .style(Button::Text)
+                    .into(),
+                PanelAnchor::Top | PanelAnchor::Bottom => button(Button::Text)
+                    .custom(vec![row![
+                        icon(
+                            "input-gaming-symbolic",
+                            self.applet_helper.suggested_size().0,
+                        )
+                        .style(Svg::Symbolic),
+                        text(match self.graphics_mode.map(|g| g.inner()) {
+                            Some(Graphics::Integrated) => fl!("integrated"),
+                            Some(Graphics::Nvidia) => fl!("nvidia"),
+                            Some(Graphics::Compute) => fl!("compute"),
+                            Some(Graphics::Hybrid) => fl!("hybrid"),
+                            None => "".into(),
+                        })
+                    ]
+                    .spacing(8)
+                    .padding([0, self.applet_helper.suggested_size().0 / 2])
+                    .align_items(Alignment::Center)
+                    .into()])
+                    .style(Button::Text)
+                    .on_press(Message::TogglePopup)
+                    .padding(8)
+                    .width(Length::Shrink)
+                    .height(Length::Shrink)
+                    .into(),
+            },
             SurfaceIdWrapper::Popup(_) => {
-                let content = match self.state {
-                    State::SelectGraphicsMode(pending_restart) => {
-                        let mut content_list = vec![
-                            radio(
-                                "Integrated Graphics",
-                                Graphics::Integrated,
-                                self.graphics_mode.map(|g| g.inner()),
-                                Message::SelectGraphicsMode,
-                            )
-                            .into(),
-                            radio(
-                                "Nvidia Graphics",
-                                Graphics::Nvidia,
-                                self.graphics_mode.map(|g| g.inner()),
-                                Message::SelectGraphicsMode,
+                let content_list = vec![
+                    button(APPLET_BUTTON_THEME)
```

Comment:
I know this wasn't your choice. This button api is extremely strange. Why do we have to reinvent every wheel?